partitions_test.go

```diff
@@ -223,3 +226,20 @@ func Test_getAllMountedDevices(t *testing.T) {
 	sort.Sort(actual)
 	assert.Equal(t, actual, sort.StringSlice(expected))
 }
+
+func TestMaybeResolveLink(t *testing.T) {
+	// If path is not a symlink, the original path should be returned.
+	tmp, err := ioutil.TempDir("", "maybeResolveLink")
+	require.Nil(t, err)
+	resolvedPath := maybeResolveLink(tmp)
+	assert.Equal(t, tmp, resolvedPath)
+
+	resolvedPath = maybeResolveLink("/dev/disk/by-partuuid")
+	assert.Equal(t, "/dev/disk/by-partuuid", resolvedPath)
+
+	tmpsym := filepath.Join(tmp, "foobar")
+	require.NoError(t, os.Symlink(tmp, tmpsym))
+	resolvedPath = maybeResolveLink(tmpsym)
```

Comment:
@kacf as we discussed. This tests if a symlink is not resolved, when it is not /dev/disk/by-partuuid. And if I did not misunderstand you, the rest would need an integration test. Good to go?

Reply:
Yes, it's just really hard to test this in a robust way without a lot of work. So let's settle for this.